File: backend/Services/Parsers/CloudleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class CloudleParser : BasicScoreResultParser
    {
        private ILogger<CloudleParser> _logger;

        public CloudleParser(ILogger<CloudleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Cloudle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName} .*? (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => Url;
        public override string Url => "https://cloudle.app/";
    }
}

File: lambdas/src/parsers/cloudle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class CloudleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Cloudle';
  readonly helpText = null;
  readonly url = 'https://cloudle.app/';
  protected readonly extraContent = this.url;
  
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `${this.gameName} .*? (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
}
